src/frontend/components/UI/AuthModal/index.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useRef } from 'react'
+import styles from './index.module.scss'
+import { ModalAnimation } from '@hyperplay/ui'
+import { WebviewTag } from 'electron'
+import { observer } from 'mobx-react-lite'
+import authState from '../../../state/authState'
+import { ethers } from 'ethers'
+import extensionStore from '../../../store/ExtensionStore'
+import onboardingStore from '../../../store/OnboardingStore'
+import walletStore from '../../../store/WalletStore'
+import { DEV_PORTAL_URL } from 'common/constants'
+
+const url = `${DEV_PORTAL_URL}/signin`
+
+const METAMASK_ALREADY_PROVIDED_ERROR_CODE = -32002
+
+const isTooManyRequestsError = (error: string) => {
+  return error.includes(METAMASK_ALREADY_PROVIDED_ERROR_CODE.toString())
+}
+
+const AuthModal = () => {
+  const webviewRef = useRef<WebviewTag>(null)
+
+  const sendRetryConnectionMessage = () => {
+    const webview = webviewRef.current
+    if (!webview) return
+    authState.setPendingSignatureRequest(false)
+    webview.send('auth:retryWalletConnection')
+  }
+
+  const handleAccountNotConnected = async () => {
+    const currentProvider = await window.api.getConnectedProvider()
+
+    if (currentProvider === 'Unconnected') {
+      window.api.openAuthModalIfAppReloads()
+      onboardingStore.openOnboarding()
+      authState.setPendingSignatureRequest(true)
+      return
+    }
+
+    // try to trigger metamask popup to connect account
+    try {
+      const provider = new ethers.BrowserProvider(window.ethereum)
+      await provider.getSigner()
+    } catch (e) {
+      // if it fails, open the popup manually
+      // since there are already requests in the queue, this will resume
+      // the connection flow after the user unlocks metamask
+      if (isTooManyRequestsError(String(e))) {
+        authState.setPendingSignatureRequest(true)
+        extensionStore.setIsPopupOpen(true)
+      }
+    }
+  }
+
+  useEffect(() => {
+    const webview = webviewRef.current
+    if (!webview) return
+
+    const handleIpcMessage = async (event: Electron.IpcMessageEvent) => {
+      switch (event.channel) {
+        case 'closeAuthModal':
+          authState.closeSignInModal()
+          break
+        case 'auth:accountConnected':
+          alert('wallet connection completed :D')
+          authState.setSignedIn()
+          authState.closeSignInModal()
+          break
+        case 'auth:accountNotConnected':
+          await handleAccountNotConnected()
+          break
+        default:
+          break
+      }
+    }
+
+    const handleDomReady = () => {
+      webview.addEventListener('ipc-message', handleIpcMessage)
+    }
+
+    webview.addEventListener('dom-ready', handleDomReady)
+
+    const qaModeListenerCleanup = window.api.handleQaModeActivated(() => {
+      authState.activateQaMode()
+      authState.openSignInModal()
```

Comment:
This we should care, if it is something only for development, staging, PTR versions, we should make a check for it depending on env for this. If this gets exposed to prod, it can be triggered.

Reply:
good call let's add another flag with a constant

Reply:
@eliobricenov we feature flag amazon game store in `src/frontend/constants.ts`
with 
`export const ENABLE_AMAZON_STORE = false`

we could do that or use the new experimental flag that flavio added in the global context provider